crates/api/src/local_user/oauth_callback.rs

```diff
@@ -0,0 +1,244 @@
+use activitypub_federation::config::Data;
+use actix_web::{
+  cookie::{Cookie, SameSite},
+  http::StatusCode,
+  web::Query,
+  HttpRequest,
+  HttpResponse,
+};
+use lemmy_api_common::{
+  claims::Claims,
+  context::LemmyContext,
+  external_auth::{OAuth, OAuthResponse, TokenResponse},
+  utils::AUTH_COOKIE_NAME,
+};
+use lemmy_api_crud::user::create::register_from_oauth;
+use lemmy_db_schema::{newtypes::ExternalAuthId, source::local_user::LocalUser, RegistrationMode};
+use lemmy_db_views::structs::{ExternalAuthView, LocalUserView, SiteView};
+use url::Url;
+
+#[tracing::instrument(skip(context))]
+pub async fn oauth_callback(
+  data: Query<OAuth>,
+  req: HttpRequest,
+  context: Data<LemmyContext>,
+) -> HttpResponse {
+  let site_view = SiteView::read_local(&mut context.pool()).await;
+
+  if !site_view.is_ok() {
+    return HttpResponse::Found()
+      .append_header(("Location", "/login?err=internal"))
+      .finish();
+  }
```

Comment:
Why not use `?` for error handling? You can customize the response headers based on the error type in `crates/utils/src/error.rs`. Or make a helper function and use `.map_err()`. In any case you shouldnt repeat the same error handling over and over, this isnt Go :P